package NYT.NChunkClient.NProto;

import "yt/ytlib/chunk_client/chunk.proto";
import "yt/ytlib/object_client/master_ypath.proto";

////////////////////////////////////////////////////////////////////////////////

message TReqLocate
{ }

message TRspLocate
{
    repeated string node_addresses = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqConfirm
{
    repeated string node_addresses = 1;
    required NYT.NChunkClient.NProto.TChunkMeta chunk_meta = 2;
    required NYT.NChunkClient.NProto.TChunkInfo chunk_info = 3;
}

message TRspConfirm
{ }

////////////////////////////////////////////////////////////////////////////////

// CreateObject extension.

message TReqCreateChunkExt
{
    extend NYT.NObjectClient.NProto.TReqCreateObject
    {
        optional TReqCreateChunkExt create_chunk = 100;
    }

    // Requested number of nodes to write.
    required int32 upload_replication_factor = 1;
 
    // Desired number of copies (including those created by background replication).
    required int32 replication_factor = 2;
 
    // Host name (without port number!) to be preferred for upload.
    // The client typically puts the name of the local host here to
    // facilitate local uploads.
    optional string preferred_host_name = 3;

    // Can this chunk be balanced to other nodes?
    optional bool movable = 4 [default = true];

    required bool vital = 5 [default = false];
}

message TRspCreateChunkExt
{
    extend NYT.NObjectClient.NProto.TRspCreateObject
    {
        optional TRspCreateChunkExt create_chunk = 100;
    }

    repeated string node_addresses = 1;
}

////////////////////////////////////////////////////////////////////////////////
